### PR TITLE
Finish anagram problem

### PR DESCRIPTION
**Pull request by PatrickHo-134**
not an optimized solution

### CODE REVIEW

File: src/assignments/phat/lesson_2_answers.clj [hunk 1]
```diff
@@ -1,3 +1,38 @@
 (ns lesson-2-answers)
 
-;;; Enter your answers in here.
+;;;;;;;;;;;;;;;;;
+;; first approach
+
+;; split each string into characters => list char-array
+;; sort list and compare
+;; if two lists equal than add the item to anagram vector
+(defn is_anagram
```
Comment (alain-picard) [inline src/assignments/phat/lesson_2_answers.clj:9]: In clojure, we tend not to use underscores to separate tokens, we use hypens.

See all the built-in functions in the clojure language.

File: src/assignments/phat/lesson_2_answers.clj [hunk 2]
```diff
@@ -1,3 +1,38 @@
...
+  [string1 string2]
+  (if (= (sort (apply list (char-array string1)))
```
Comment (alain-picard) [inline src/assignments/phat/lesson_2_answers.clj:11]: strings are sequences.  Try:

(sort string)

File: src/assignments/phat/lesson_2_answers.clj [hunk 3]
```diff
@@ -1,3 +1,38 @@
...
+;; not done
+
+;;;;;;;;;;;;;;;;;;
+;; second solution
+
+;; convert each item into a set
+(defn find_anagram 
+  [vector] 
+  (set 
+   (map set (vals (group-by set vector)))))
```
Comment (alain-picard) [inline src/assignments/phat/lesson_2_answers.clj:26]: Nice attempt, but unfortunately buggy:
```
(find_anagram ["veer" "lake" "item" "kale" "mite" "ever" "everever"])
;; =>> #{#{"ever" "everever" "veer"} #{"kale" "lake"} #{"item" "mite"}}
```
Can you spot the problem?

Comment (PatrickHo-134) [reply src/assignments/phat/lesson_2_answers.clj:26]: Yes. I have noticed this problem. 

Thanks for your recommendation above with (sort string), I manage to fix this bug by group-by sorted list of characters.

The latest version of this function would be:

(defn find-anagram 
  [vector] 
  (set 
   (filter #(>= (count %) 2) 
           (set (map set (vals (group-by sort vector)))))))

I will make it in the next pull request.

Best regards

**Pull request merged**